File: frontend/src/models/Request.ts
import type { RequestContact } from "./RequestContact";

export interface Response {
  contacts?: RequestContact[];
  request: Request;
}

export interface Request {
  id?: number;
  brand: string;
  requestType: string;
  submissionDate: string;
  primaryContactName: string;
  primaryContactPhone: string;
  contacts?: RequestContact[];
}

File: backend/src/main/java/com/oscar/backend/model/Request.java
package com.oscar.backend.model;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table("requests")
public class Request {
    @Id
    private Integer id;
    private String brand;
    private String requestType;
    private LocalDate submissionDate;
    private String primaryContactName;
    private String primaryContactPhone;
}
